src/fides/api/ops/service/connectors/saas_query_config.py

```diff
@@ -51,42 +51,76 @@ def __init__(
         self.data_protection_request = data_protection_request
         self.privacy_request = privacy_request
         self.action: Optional[str] = None
+        self.current_request: Optional[SaaSRequest] = None
 
-    def get_request_by_action(self, action: str) -> Optional[SaaSRequest]:
```

Comment:
The `get_request_by_action` function was replaced by `get_read_requests_by_identity` and `get_erasure_request_by_action`. Another change that I had to make to facilitate with the different types returned by the read vs update and delete.